src/main/java/io/vertx/core/http/impl/Http1xConnectionBase.java

```diff
@@ -40,13 +50,36 @@
   }
 
   void handleWsFrame(WebSocketFrame msg) {
+    WebSocketFrameInternal frame = decodeFrame(msg);
     WebSocketImplBase<?> w;
     synchronized (this) {
       w = webSocket;
     }
     if (w != null) {
-      w.context.execute(msg, w::handleFrame);
+      w.context.execute(frame, w::handleFrame);
+    }
+  }
+
+  private WebSocketFrameInternal decodeFrame(io.netty.handler.codec.http.websocketx.WebSocketFrame msg) {
```

Comment:
If each case is equally possible, it's better to use a
`static final IdentityHashMap<Class<?>,FrameType` initialized with the handled types, given that `System::identityHashCode` is an intrinsic, msg::getClass too and `Class` should be singleton in the same class loader AFAIK!

Reply:
thanks, feel free to do a PR because it is another concern than this PR and the code was just moved around :-)